src/XML/src/FMI3/fmi3_xml_parser_context_impl.h

```diff
@@ -28,23 +28,50 @@
 extern "C" {
 #endif
 
+/** Flag for current XML file being handled. */
+// TODO: Might be more suited in parser.h?
+typedef enum fmi3_xml_type_t {
+    fmi3_xml_type_modelDescription,
+    fmi3_xml_type_terminalAndIcons
+} fmi3_xml_type_t;
+
+typedef struct fmi3_xml_modelDescription_element_handle_map_t fmi3_xml_modelDescription_element_handle_map_t;
+typedef struct fmi_xml_termIcon_element_handle_map_t fmi_xml_termIcon_element_handle_map_t;
 typedef struct fmi3_xml_element_handle_map_t fmi3_xml_element_handle_map_t;
 
+typedef int (*fmi3_xml_modelDescription_element_handle_ft)(fmi3_xml_parser_context_t* context, const char* data);
+typedef int (*fmi3_xml_termIcon_element_handle_ft)(fmi3_xml_parser_context_t* context, const char* data);
 typedef int (*fmi3_xml_element_handle_ft)(fmi3_xml_parser_context_t* context, const char* data);
 
+struct fmi3_xml_modelDescription_element_handle_map_t {
+    const char* elementName;
+    fmi3_xml_modelDescription_element_handle_ft elementHandle;
+    fmi3_xml_elm_modelDescription_enu_t elemID;
+};
+
+struct fmi_xml_termIcon_element_handle_map_t {
+    const char* elementName;
+    fmi3_xml_termIcon_element_handle_ft elementHandle;
+    fmi_xml_elm_termIcon_enu_t elemID;
+};
+
 struct fmi3_xml_element_handle_map_t {
     const char* elementName;
     fmi3_xml_element_handle_ft elementHandle;
-    fmi3_xml_elm_enu_t elemID;
+    fmi3_xml_elm_t elemID;
 };
 
+
 jm_vector_declare_template(fmi3_xml_element_handle_map_t)
 
 /**
  * Struct for saving and accessing data between element handlers.
  */
 struct fmi3_xml_parser_context_t {
 
+    /* Flag for XML being handled, MUST BE FIRST */
```

Comment:
I think we could add in this comment that it is either `...` or `...` depending on what we are parsing.

Reply:
Expanded the comment